src/litdata/streaming/reader.py

```diff
@@ -291,6 +301,12 @@ def __getstate__(self) -> Dict[str, Any]:
         state["_prepare_thread"] = None
         return state
 
+    def __del__(self) -> None:
+        if self._prepare_thread and not self._prepare_thread._has_exited:
+            self._prepare_thread.force_stop()
```

Comment:
The force stop is needed because when `if self._pre_download_counter < self._max_pre_download` is True in `PrepareChunksThread`, the `_END_TOKEN` will never be received.